src/main/kotlin/org/jetbrains/research/tasktracker/tracking/activity/ActivityTracker.kt

```diff
@@ -0,0 +1,127 @@
+package org.jetbrains.research.tasktracker.tracking.activity
+
+import com.intellij.execution.ExecutionListener
+import com.intellij.execution.ExecutionManager
+import com.intellij.execution.process.BaseProcessHandler
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.ide.IdeEventQueue
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.IdeActions
+import com.intellij.openapi.actionSystem.ex.AnActionListener
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.util.messages.MessageBusConnection
+import org.jetbrains.research.tasktracker.tracking.logger.ActivityLogger
+import java.awt.AWTEvent
+import java.awt.event.KeyEvent
+
+class ActivityTracker(project: Project) {
+    private val activityLogger: ActivityLogger
+    private val messageBusConnections: MutableList<MessageBusConnection>
+    private var trackingDisposable: Disposable? = null
+
+    init {
+        activityLogger = ActivityLogger(project)
+        messageBusConnections = mutableListOf()
+    }
+
+    // TODO: add config to select activities to track
+    fun startTracking() {
+        trackingDisposable = Disposer.newDisposable()
+        listenActions()
+        listenKeyboard()
+        listenExecution()
+    }
+
+    fun stopTracking() {
+        messageBusConnections.forEach { it.disconnect() }
+        trackingDisposable?.let { Disposer.dispose(it) }
+    }
+
+    private fun listenActions(listenShortcut: Boolean = true) {
+        val actionListener = object : AnActionListener {
+            override fun beforeActionPerformed(anAction: AnAction, event: AnActionEvent) {
+                val actionId = ActionManager.getInstance().getId(anAction) ?: return
+                if (listenShortcut && !movingActions.contains(actionId) && !typingActions.contains(actionId)) {
+                    val input = event.inputEvent
+                    if (input is KeyEvent) {
+                        activityLogger.log(Type.Shortcut, eventToShortcutInfo(input))
+                    }
```

Comment:
I'd prefer to create a function 'fun extractShortcut(...): String?' and use it here